src/java.base/share/classes/java/nio/file/package-info.java

```diff
@@ -82,6 +82,12 @@
  * java.nio.file.spi.FileSystemProvider provider} implementations is provider
  * specific. </p>
  *
+ * <h2><a id="directories">Operations on Directories</a></h2>
+ * <p> Many operations which can be performed on regular files can also be
+ * performed on directories. It is highly implementation specific, however, as
+ * to whether such operations will succeed. These operations will either work
+ * or throw an {@code IOException}. </p>
+ *
```

Comment:
I'm not sure about this wording or section title. I think the point is that a file path can locate any time of file, be it a regular file, directory, sym link, or something else. There is nothing to prevent attempting a file operation on any file but it may be that some operations aren't applicable or supported on some file types. It may be that this part should be a separate issue/PR as it's nothing to do with Files.mismatch.